Makefile

```diff
@@ -480,6 +480,7 @@ mocks: ## Generate mocks
 	${GOPATH}/bin/mockgen -destination=pkg/helm/mocks/download.go -package=mocks -source "pkg/helm/download.go"
 	${GOPATH}/bin/mockgen -destination=pkg/aws/mocks/ec2.go -package=mocks -source "pkg/aws/ec2.go"
 	${GOPATH}/bin/mockgen -destination=pkg/aws/mocks/snowballdevice.go -package=mocks -source "pkg/aws/snowballdevice.go"
+	${GOPATH}/bin/mockgen -destination=pkg/providers/nutanix/mock_client.go -package=nutanix -source "pkg/providers/nutanix/client.go"
```

Comment:
Was this intentionally added to the `nutanix` package making it part of its API?

Reply:
This is just to add generate mocks. This doesn't add `nutanix` package to any APIs

Reply:
Not quite. Its generating a file used in `go build` with the package name `nutanix` which means the mock data structures are part of the `nutanix` package API. I'm not sure if that was on purpose or accidental. Normally, you'd generate a `mock_client_test.go` (`_test` etc) or put it in a separate `mocks` package (which I'm not a fan of). 

What's the intent here? Having it as part of the package API seems odd but there may be a use-case if its needed externally.

Reply:
That's a fair point given the mock is being checked in rather than being generated on the fly. I've renamed the generated mock file with a `_test` suffix. Not a fan of a separate mocks package either.